### ISSUE TITLE
Feature request: Disallow toggling the negative/positive sign with '+' and '-' keypress

### ISSUE THREAD

**Issue by generatorr** [autoNumeric/autoNumeric#709]
### Current behavior
The +/- sign flips when the user presses the "+" or the "-" key.
- In an empty field, hit the "+" key and a "-" appears. Hit the "+" key again and the "-" sign vanishes
- In a filled field with positive value, hit the "+" key and a "-" appears (value is negative). Hit the "+" key again and the "-" sign vanishes (value is positive again)
- "+" and "-" keys have the same functionality - hitting the "-" key several times flips the sign.

This is quite unexpected for our users and we consider this a bug. I don't know of any other software where "+" and "-" sign behave in a similar way.
Especially for fast-typing users it is totally unexpected that when pressing the keys "+123" they get the value "-123". That's why I consider this a bug report rather than a feature request

### Expected behavior
- Hitting the "+" key should make the number positive (if allowed by minValue and maxValue)
- Hitting the "-" key should make the number negative (if allowed by minValue and maxValue)
- The "Flip the sign" behaviour vs. "+ means positive, - means negative" behaviour could be controlled by a boolean setting e.g. "setFlipSignOnPlusMinusKey"
- setFlipSignOnPlusMinusKey(true) - default; "+" and "-" key flip the sign
- setFlipSignOnPlusMinusKey(false) - "+" and "-" keys are only allowed at the beginning (or end, depending on defined format). No sign flipping.

### Steps to reproduce the problem
We are using PrimeFaces, which again is using AutoNumeric. Please see the PrimeFaces Showcase page:
- use [PrimeFaces Showcase for InputNumber](https://primefaces.org/showcase/ui/input/inputNumber.xhtml)
- Type the keys "+123" in any of the InputNumber fields and you will get the value "-123" in the field.

### Link to live example (ie. [Codepen](http://codepen.io/pen/))
https://primefaces.org/showcase/ui/input/inputNumber.xhtml?jfwid=057bf

Location is:

./src/AutoNumeric.js

```js
   9021         if ((this.eventKey === '-' || this.eventKey === '+') && this.settings.isNegativeSignAllowed) {
   9022             // Here, the left and right parts have been normalized already, hence the minus sign usage
   9023             if (left === '' && AutoNumericHelper.contains(right, '-')) {
   9024                 // The value is originally negative (with a trailing negative sign)
   9025                 right = right.replace('-', '');
   9026             } else if (AutoNumericHelper.isNegativeStrict(left, '-')) {
   9027                 // The value is originally negative (with a leading negative sign)
   9028                 // Remove the negative sign, effectively converting the value to a positive one
   9029                 left = left.replace('-', ''); //TODO replace with '+' if `showPositiveSign` too?
   9030             } else {
   9031                 // The value is originally positive, so we toggle the state to a negative one (unformatted, which means even with a trailing negative sign, we add the minus sign on the far left)
   9032                 left = `${this.settings.negativeSignCharacter}${left}`;
   9033             }
   9034 
   9035             this._setValueParts(left, right);
   9036 
   9037             return true;
   9038         }
```

**Comment by generatorr** [comment]
The original issue report in the Primefaces Forum: https://forum.primefaces.org/viewtopic.php?f=3&t=65159

**Comment by AlexandreBonneau** [comment]
This is fixed in `4.8.0`, with the addition of a new option `negativePositiveSignBehavior`.

Do note that the default behavior is changed in this version, where toggling between the positive and negative value with the '-' or '+' keys is not activated by default. If you want to use the previous behavior, please set `negativePositiveSignBehavior: AutoNumeric.options.negativePositiveSignBehavior.toggle` in your options' configuration.
